packages/invoices/src/InboundShipment/DetailView/DetailView.tsx

```diff
@@ -1,60 +1,108 @@
 import React, { FC } from 'react';
 import { useParams } from 'react-router';
 import {
-  Column,
+  useQuery,
   TableProvider,
   createTableStore,
   useOmSupplyApi,
-  useDocument,
   useColumns,
   GenericColumnKey,
   getNotePopoverColumn,
   getRowExpandColumn,
   useDialog,
   DialogButton,
   useTranslation,
-  Item,
+  useMutation,
+  useQueryClient,
 } from '@openmsupply-client/common';
 
-import { InboundAction, reducer } from './reducer';
 import { getInboundShipmentDetailViewApi } from './api';
 import { Toolbar } from './Toolbar';
 import { Footer } from './Footer';
 import { AppBarButtons } from './AppBarButtons';
 import { SidePanel } from './SidePanel';
 import { GeneralTab } from './GeneralTab';
 import { InboundLineEdit } from './modals/InboundLineEdit/InboundLineEdit';
-
-import { isInboundEditable } from '../../utils';
-import { InboundShipmentItem } from '../../types';
+import { inboundLinesToSummaryItems } from './reducer/reducer';
+import {
+  getNextInboundStatus,
+  isInboundEditable,
+  placeholderInbound,
+} from '../../utils';
+import {
+  InboundShipmentItem,
+  InboundShipment,
+  Invoice,
+  OutboundShipmentSummaryItem,
+} from '../../types';
 
 const useDraftInbound = () => {
+  const queryClient = useQueryClient();
   const { id } = useParams();
   const { api } = useOmSupplyApi();
+  const queries = getInboundShipmentDetailViewApi(api);
 
-  const { draft, save, dispatch, state } = useDocument(
-    ['invoice', id],
-    reducer,
-    getInboundShipmentDetailViewApi(api)
-  );
+  const onChangeSortBy = () => {};
+
+  const { data } = useQuery(['invoice', id], () => {
+    return queries.onRead(id);
+  });
+
+  const draft = data
+    ? { ...data, items: inboundLinesToSummaryItems(data?.lines ?? []) }
+    : placeholderInbound;
+
+  const { mutateAsync } = useMutation(queries.onUpdate, {
+    onMutate: async (patch: Partial<InboundShipment>) => {
+      await queryClient.cancelQueries(['invoice', id]);
+
+      const previousInbound: Invoice = queryClient.getQueryData([
+        'invoice',
+        id,
+      ]);
+
+      queryClient.setQueryData(['invoice', id], {
+        ...previousInbound,
+        ...patch,
+      });
+
+      return { previousInbound, patch };
+    },
+    onSettled: () => queryClient.invalidateQueries(['invoice', id]),
```

Comment:
wondering if this is required, since `setQueryData` is used on mutate and `onError`. The refetch _should_ just be setting the cached value to be the same as that returned by the query

Reply:
Oops sorry I didn't address this. We could just assume that the cache update was correct. I'm not sure I see the harm in refetching, though, just incase? For the case of adding items, this should do the side effects such as getting new pricing totals etc which maybe we should be doing in the optimistic update, also? I suppose it's also a way to update values all throughout the invoice if others are updating it also which could be useful? (also, harmful?)

Reply:
Fine to refetch, as long as we debounce - currently there are a lot of mutate & query combos getting fired off! in most cases a quick cache update would be exactly the same as the response from the query, but you're right, it might not be always, and is safer to refetch.